xla/stream_executor/command_buffer.h

```diff
@@ -47,90 +48,22 @@ class Stream;
+// command index to node dependencies in the implementation.
 class CommandBuffer {
  public:
-  // Execution scope enables fine-grained synchronization scopes inside
-  // commands buffers. Implementation is very backend-specific and for CUDA/ROCM
-  // backends it's implemented as DAG edges. By default all commands launched in
-  // the `kDefaultExecutionScope` execution scope.
-  //
-  // Example #1: independent execution scopes and independent barriers
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A                        D
-  //          B                        E
-  // ----- barrier -----      ----- barrier -----
-  //          C                        F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before C.
-  //   (2) Commands D and E can run concurrently and must complete before F.
-  //   (3) There is no syncrhonization between execution scopes, and commands
-  //       from different execution scopes can execute concurrently with each
-  //       other as long as they satisfy constraints of their respective
-  //       execution scopes.
-  //
-  //
-  //
-  // Example #2: dependencies between scopes and inter-scope barriers
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A                        D
-  //          B                        E
-  // ----------------- barrier ------------------
-  //          C                        F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before
-  //       C and F.
-  //   (2) Commands D and E can run concurrently and must complete before
-  //       C and F.
-  //   (3) Commands C and F can run concurrently.
-  //   (4) All commands before a shared barrier (in both excecution scopes)
-  //       should complete before any command after a berrier starts execution.
-  //
-  //
-  //
-  // Example #3: one-directional barriers between execution scopes
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A
-  //          B
-  // ----- barrier -----               D
-  //          C            \           E
-  //                           ----- barrier -----
-  //                                   F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before
-  //       C and F.
-  //   (2) Commands D and E can run concurrently and must complete before
-  //       F (does not synchronize with C).
-  //   (3) Commands C and F can run concurrently.
-  //
-  //  This is a more fine-grained barrier than in example #2: it enforces
-  //  synchronization from execution scope #0 to execution scope #1 but no
-  //  synchronization in other direction. For CUDA/ROCM backend it has the same
-  //  semantics as stream wait operation.
-  //
-  TSL_LIB_GTL_DEFINE_INT_TYPE(ExecutionScopeId, uint64_t);
-  static constexpr auto kDefaultExecutionScope = ExecutionScopeId(0);
+  typedef int64_t Index;
```

Comment:
Can you also document what's Index and how exactly it gets assigned to commands, just a few lines to make it clear why it's here

Reply:
(just found that it seems to be assigned by the user, just make it clear that this is the contract)